module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -1010,14 +1011,16 @@ public function editAction()
         }
 
         // Get saved favorites for selected list (or all lists if $listID is null)
-        $userResources = $user->getSavedData($id, $listID, $source);
+        $userResourceService = $this->getDbService(UserResourceServiceInterface::class);
+        $userResources = $userResourceService->getFavoritesForRecord($id, $source, $listID, $user);
         $savedData = [];
         foreach ($userResources as $current) {
+            $currentList = $current->getUserList();
```

Comment:
`null` is a legal return value for `getUserList()`, do we need to take that into account here?